wal/wal_test.go

```diff
@@ -269,3 +271,43 @@ func TestWALTruncate(t *testing.T) {
 		require.Equal(t, 1, numRecords)
 	})
 }
+
+func TestWALCloseTimeout(t *testing.T) {
+	dir := t.TempDir()
+	w, err := Open(
+		log.NewNopLogger(),
+		prometheus.NewRegistry(),
+		dir,
+	)
+	require.NoError(t, err)
+	defer w.Close()
+
+	closed := atomic.Bool{}
+	closed.Store(false)
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
```

Comment:
nit: rather than setting up the cancel infra and running the wal yourself, I think it would be cleaner to call `w.RunAsync` here and defer `w.Close`. It's fine to fail the test due to timeout.

Reply:
ah of course, good call will fix